src/GitHub.App/ViewModels/PullRequestDetailViewModel.cs

```diff
@@ -426,38 +413,60 @@ string GetBranchDisplayName(string targetBranchLabel)
 
         IObservable<Unit> DoCheckout(object unused)
         {
-            IObservable<Unit> operation = null;
-
-            switch (CheckoutMode)
+            return Observable.Defer(async () =>
             {
-                case CheckoutMode.NeedsPull:
-                    operation = pullRequestsService.Pull(repository);
-                    break;
-                case CheckoutMode.Fetch:
-                    operation = pullRequestsService
+                var localBranches = await pullRequestsService.GetLocalBranches(repository, Model).ToList();
+
+                if (localBranches.Count > 0)
```

Comment:
Actually scratch that, it's an observable so returning many is no different. I can use `FirstOrDefaultAsync` however.